### PR TITLE
Support Music Shuffle

### LINKED ISSUE #69: Add Music Shuffle
Ideally have two options: 
- Shuffle all music together 
- Shuffle areas and metroid/bosses separately

### PR DESCRIPTION
**Pull request by dyceron**
Takes a dictionary of tracks as input and replaces the data with a new specified track

Fixes #69

### CODE REVIEW

File: src/open_samus_returns_rando/specific_patches/cosmetic_patches.py [hunk 1]
```diff
@@ -33,3 +34,10 @@ def lua_cosmetics(configuration: dict) -> str:
     }
 
     return lua_util.replace_lua_template("cosmetics.lua", replacement)
+
+
+def music_shuffle(editor: PatcherEditor, configuration: dict) -> None:
+    for original, new in configuration["music_shuffle_dict"].items():
+        original_track = f"sounds/streams/music/{original}.bcwav"
+        new_track = editor.get_raw_asset(f"sounds/streams/music/{new}.bcwav")
+        editor.replace_asset(original_track, new_track)
```
Comment (Miepee) [inline src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:43]: This currently has some unexpected behaviour with something like
```json
"k_crateria99": "boss_go",
"boss_go": "k_crateria99"
```

File: src/open_samus_returns_rando/specific_patches/cosmetic_patches.py [hunk 2]
```diff
@@ -33,3 +34,14 @@ def lua_cosmetics(configuration: dict) -> str:
     }
 
     return lua_util.replace_lua_template("cosmetics.lua", replacement)
+
+
+def music_shuffle(editor: PatcherEditor, configuration: dict) -> None:
```
Comment (Miepee) [inline src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:39]: can you early return? something like
```py
if configuration["music_shuffle_dict"] == {}:
  return
```

File: src/open_samus_returns_rando/files/schema.json [hunk 3]
```diff
@@ -497,6 +497,17 @@
                             }
                         }
                     }
+                },
+                "music_shuffle_dict": {
+                    "description": "A dictionary mapping music tracks to the output of music mode.",
```
Comment (ThanatosGit) [inline src/open_samus_returns_rando/files/schema.json:502]: I don't get what this has to do with `music mode`. 
It swaps one track with the other.

Comment (dyceron) [reply src/open_samus_returns_rando/files/schema.json:502]: I don't understand the confusion.

Comment (Miepee) [reply src/open_samus_returns_rando/files/schema.json:502]: Music mode has no effect here and is completely irrelevant. A more accurate description would be something along the lines of "A dictionary mapping, where the key is a track that gets swapped with the value"

Comment (dyceron) [reply src/open_samus_returns_rando/files/schema.json:502]: Fwiw this was made before the rdv pr, so this is just a small wording error on my part. I'll fix it.

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**